src/git/doGitPush.ts

```diff
@@ -1,6 +1,22 @@
+import getOptionalBooleanFlagArgument from "../utils/getOptionalBooleanFlagArgument.js";
 import spawn from "../utils/spawn.js";
 import gitLogger from "./utils/gitLogger.js";
 
-export default async function (remote = "origin", ...refs: string[]) {
-  await spawn("git", ["push", remote, ...refs], gitLogger);
+type GitPushFlags = {
+  force: boolean;
+};
+
+const defaultGitPushFlag = { force: false };
+export default async function (
+  { force }: GitPushFlags | undefined = defaultGitPushFlag,
```

Comment:
If this needs to be a breaking change, can the refs be changed to not have spread but an explicit string[] so that push flag can be added at the end?

Reply:
Can't the force flags param be made optional and the default values assigned inside the function - so that it can be skipped when not needed?

Reply:
> If this needs to be a breaking change, can the refs be changed to not have spread but an explicit string[] so that push flag can be added at the end?

We can, and honestly, I'm open to it :)

I see two reasons against it, however:
 1. The parameter used to call it would be in a different order than one of the function signatures, this could appear as not immediately transparent to the end-user
 2. `gitPush(,,'myBranch1', 'myBranch2')` is IMO leaner than `gitPush(,'myBranch1', 'myBranch2')`

My thought process for this rework was as follows:
 1. Handle all the flags we want to support
 2. Handle the positional arguments)
 3. Make the function looks like the git push command as much as possible (without adding support for all flags)

Reply:
For me, `undefined` as a param always seemed out of place. Since we're doing a breaking change should we take the opportunity to change the code to use an object for the param instead? It would allow us to add more flags in the future without doing more breaking changes:

```typescript
interface GitPushParams {
  force?: boolean;
  remote?: string;
  refs?: string[];
  // ...
}
const defaultPushParams: GitPushParams = {{remote: 'origin', force: false, refs: []}};
export default async function (params: GitPushParams) {
  const {remote, force, refs} = {...defaultPushParams, ...params};
  // ...
}
```

- `gitPush('myRemote')` -> `gitPush({remote: 'myRemote'})`
- `gitPush('myRemote', 'myRef')` -> `gitPush({remote: 'myRemote', refs: ['myRef']})`
- `gitPush(undefined, 'myRef')` -> `gitPush({refs: ['myRef']})`

And if you want to force push -> `gitPush({remote: 'myRemote', force: true})`


We could also do shenanigans to be backward compatible while allowing `force` only on the new way of doing things, but I think it would be more confusing in the long run:
`export default async function (paramsOrRemote: GitPushParams | string, ...refs: string[]) {`

Reply:
Alright, let's go to with the 'object' 👍